addons/mail/models/mail_message.py

```diff
@@ -1090,3 +1093,18 @@ def _invalidate_documents(self, model=None, res_id=None):
 
     def _get_search_domain_share(self):
         return ['&', '&', ('is_internal', '=', False), ('subtype_id', '!=', False), ('subtype_id.internal', '=', False)]
+
+    def _is_empty_message(self):
+        self.ensure_one()
+        is_body_empty = not self.body or is_html_empty(self.body)
+        is_subtype_description_empty = not self.subtype_id or not self.subtype_id.description or is_html_empty(html_sanitize(self.subtype_id.description, strip_style=True, strip_classes=True))
```

Comment:
I am unsure a vboi description qualifies for "empty". This is just a trick done by frontend but the field is just an additional information compared to the name.

Having business logic based on it is highly strange, especially users could tweak it. If I set a description, should we go back to all "archived" messages and set them back as "not empty" ? I don't think so ... this field was never intended to define a "void subtype".

Reply:
This is how it works though, we have to add the condition or we are going to ignore a lot of messages.

Pretty much all subtypes that do have a description are used on messages with empty "everything else". This is the only thing that shows in chatter for them.

Changing (adding or removing) those kind of subtype description is like a configuration error in their current usage.

Reply:
> Changing (adding or removing) those kind of subtype description is like a configuration error in their current usage.

According to the help: 'Description that will be added in the message posted for this subtype. If void, the name will be added instead.' It was always thought to be used that way: name is a fallback to description. That's strange to base code on a field being void when it is just a substitute to the name. Seems like a decision from UI that adds constraints for the model and that makes no sense.

Reply:
"A lot of subtypes that do have a description are used on messages with empty "everything else". This is the only thing that shows in chatter for them."

When do we have that ? Seems like noise to me ?

Reply:
"Invoice Created" is an example. This tracking is purely based on the subtype, there is no message data.
It's actually a good thing, this gives more meaning to the message and the text is actually translatable/translated as opposed to just hardcoding some value into message body like some other models do.

The problem if we actually fallback on the subtype name is that then it's never empty... And we can no longer differentiate actually empty from just "using subtype as content". 

Unless we check for specific subtype instead? Empty everything else and subtype is  `mt_comment` or `mt_note` -> consider empty? And in all other cases, never empty?

Reply:
Arf yes, that "creation subtype" thingy ... indeed now we don't have tracking values at creation time. Well it's ok for stable I think then, at least it makes the code compatible with frontend. I think the' whole meaning of "emptying / deleting" messages could be cleaned in master to remove code and strange conditions. Keeping the old message is probably required (for example at least for Message ID and mailgateway, if someone replies, if any email has been sent) but we could think of an active-like flag. Maybe it is costly to add a column on message, but could be worth a shot ...